tests/test_trainer_callback.py

```diff
@@ -129,10 +129,7 @@ def get_expected_events(self, trainer):
                 expected_events += ["on_step_begin", "on_step_end"]
                 if step % trainer.args.logging_steps == 0:
                     expected_events.append("on_log")
-                if (
-                    trainer.args.evaluation_strategy == EvaluationStrategy.STEPS
-                    and step % trainer.args.eval_steps == 0
-                ):
+                if trainer.args.evaluation_strategy == "steps" and step % trainer.args.eval_steps == 0:
```

Comment:
Why is this change necessary? This should have been cast to `EvaluationStrategy.STEPS` at init.

Reply:
As you can see from https://github.com/huggingface/transformers/pull/9800/commits/2195c1fd4929f9ef16600d60d688b95d955356d5 tests - one under `tests` and several under `examples` tests fail with this PR. I am not quite sure why. 

That was one attempt, and then one more above: https://github.com/huggingface/transformers/pull/9800#issuecomment-767908408